less_26.JS

```diff
@@ -0,0 +1,104 @@
+// *********************************** LESS 26 ********************************************
+
+// # 1
+
+const url = new URL('https://dummyjson.com/products');
+
+class ProductsService {
+    constructor(url) {
+        this.url = url
+    }
+    
+    getAllProduct = async () => {
+        await fetch(url)
+            .then(res => res.json())
+            .then((data) => console.log(data))
+            .catch((err) => {
+                console.log(err);
+        });
+    }
+
+    getSingleProduct = async (idnum) => {
+        await fetch(`${url}/${idnum}`)
+            .then(res => res.json())
+            .then((data) => console.log(data))
+            .catch((err) => {
+                console.log(err);
+        });
+    }
+
+    searchProducts = async (prod) => {
+        const params = new URLSearchParams();
+        params.append("q", prod);
+        const reps = await fetch(`${url}/search?${params}`);
+        const data = await reps.json();
+        console.log(data);
+    }
+    
+    getAllProductsCategories = async () => {
+        await fetch(`${url}/categories`)
+            .then(res => res.json())
```

Comment:
try to stick to 1 approach
in prev method you used async await so pick one and use it always across a class